### DISCUSSION TITLE
Easiest way to get all courses

### DISCUSSION THREAD

**Discussion by MarkusvonStaden** [ucfopen/canvasapi#585] [Q&A] [answered]
Hi,
Is it possible to loop over all the courses without being enrolled in them (as an admin user of corse)?

**Comment by bennettscience** [comment, marked as answer]
Are you looking for all the courses in an account? If you have admin access, you can do `canvas.get_courses()` ([ref](https://canvasapi.readthedocs.io/en/stable/canvas-ref.html)). That calls the [top-level](https://canvas.instructure.com/doc/api/courses.html#method.courses.index) course endpoint for your Canvas instance. I would recommend using some `kwarg` filters to limit the number of courses you actually get back.

**Reply by MarkusvonStaden** [reply]
No, I need to get all courses available on canvas. As a workaround I generated a csv file from the admin settings and used that to get all course ids and loop over that csv file. Not ideal though, since i need to do something and cannot automate the entire task. But I couldn't find a way directly with their api either.

**Reply by bennettscience** [reply]
Oops, sorry - that's my fault. I gave the wrong answer. You need to use `account.get_courses()` to get everything. _Most_ Canvas instances have a root account with the id `1`. You can do:

```python
# Imports, etc
canvas = Canvas(URL, KEY)
account = canvas.get_account(1)

all_courses = account.get_courses()
```

You can also get all of the available account IDs with `canvas.get_accounts()` and then target specific subaccounts. Sorry about that.

**Reply by MarkusvonStaden** [reply]
That works, thanks a lot!
